packages/fx-core/src/core/middleware/utils/debug/taskMigrator.ts

```diff
@@ -94,32 +94,62 @@ export async function migrateTransparentPrerequisite(
 }
 
 export async function migrateTransparentLocalTunnel(context: DebugMigrationContext): Promise<void> {
-  for (const task of context.tasks) {
+  let index = 0;
+  while (index < context.tasks.length) {
+    const task = context.tasks[index];
     if (
       !isCommentObject(task) ||
       !(task["type"] === "teamsfx") ||
       !(task["command"] === TaskCommand.startLocalTunnel)
     ) {
+      ++index;
       continue;
     }
 
     if (isCommentObject(task["args"])) {
-      if (task["args"]["ngrokArgs"] === TaskDefaultValue.startLocalTunnel.ngrokArgs) {
-        task["args"] = generateLocalTunnelTaskArgs(context);
-        const comment = `{
-          // Start the local tunnel service to forward public URL to local port and inspect traffic.
-          // See https://aka.ms/teamsfx-tasks/local-tunnel for the detailed args definitions.
-        }`;
-        const comments = task[Symbol.for("before:label") as CommentSymbol];
-        comments?.splice(0, comments?.length ?? 0);
-        assign(task, parse(comment));
-      } else {
-        // TODO: use shell task and js script to start global ngrok
+      if (typeof task["args"]["ngrokArgs"] === "string") {
+        const portNumber = getNgrokPort(task["args"]["ngrokArgs"]);
+        if (portNumber) {
+          task["args"] = generateLocalTunnelTaskArgs(context, portNumber);
+          const comment = `{
+            // Start the local tunnel service to forward public URL to local port and inspect traffic.
+            // See https://aka.ms/teamsfx-tasks/local-tunnel for the detailed args definitions.
+          }`;
+          const comments = task[Symbol.for("before:label") as CommentSymbol];
+          comments?.splice(0, comments?.length ?? 0);
+          assign(task, parse(comment));
+          ++index;
+          continue;
+        }
       }
     }
+
+    const comment = `{
+          // Teams Toolkit now uses Dev Tunnel as default tunnel solution.
+          // See https://aka.ms/teamsfx-tasks/local-tunnel for more details.
+          // If you still prefer to use ngrok, please refer to https://aka.ms/teamsfx-tasks/customize-tunnel-service to learn how to use your own tunnel service.
+        }`;
+    const newTask = assign(parse(comment), {
+      label: task["label"],
+      type: "shell",
+      command:
+        "echo 'Teams Toolkit now uses Dev Tunnel as default tunnel solution. For manual updates, see https://aka.ms/teamsfx-tasks/local-tunnel.'; exit 1;",
```

Comment:
The command separator `;` does not work in windows cmd. Would that be a potential issue if user's default shell is cmd?

Reply:
Thx, change the command to be "echo 'Teams Toolkit now uses Dev Tunnel as default tunnel solution. For manual updates, see https://aka.ms/teamsfx-tasks/local-tunnel.' && exit 1" and specify the shell to be cmd.